Updated video embeds

AdarshBhandari15 opened the pull request:
Fixed issue with video embeds not always being generated. Added a new case for if the video is of type 'player-video' in which a new embed code is generated for this different type of video. However, this has created a new issue (#35) that there can now be duplicate videos if there are multiple versions of the same video with different file types.

To test this feature once the publichtml repository is updated to reflect this plugin change:
1. Create a new entry with a tinydesk story that uses a player-video type video (ex: storyId='579982124')
2. Save the entry and give it the category "Illinois Soul"
3. Visit localhost/illinoissoul and scroll down to see the new entry
4. Click on the entry and see that there is only 1 video that appears along with the rest of the article correctly formatted.

AdarshBhandari15 committed:
Updated video embeds

areynold commented on libraries/mapping/publish_form_mapper.php:
```diff
@@ -659,6 +659,8 @@ private function get_video_streaming($asset, $profile): ?array
 
         if ($profile === 'player-video') {
+            $poster = '';
+            $video_url = $asset_current->enclosures[0]->href;
```
Check your variable names. I'm showing $asset_current undefined at this location

areynold commented on libraries/mapping/publish_form_mapper.php:
```diff
@@ -667,19 +669,38 @@ private function get_video_streaming($asset, $profile): ?array
             $video['enclosures'] = $enclosures;
 
-            if (!empty($asset->images)) {
+            if (!empty($asset_current->images)) {
```
Same as line 663. $asset_current undefined

areynold commented on libraries/mapping/publish_form_mapper.php:
```diff
@@ -667,19 +669,38 @@ private function get_video_streaming($asset, $profile): ?array
+                foreach ($asset_current->images as $v_image) {
+                    if (in_array('thumbnail', $v_image->rels)) {
+                        $v_image_id = $this->extract_asset_id($v_image->href);
+                        $v_image_asset = $story->assets->{$v_image_id};
```
$story is undefined here

AdarshBhandari15 committed:
Fix video aspect ratios

AdarshBhandari15 commented:
Fixed variable problems listed above and also fixed duplicate videos appearing from issue #35. Also, fixed issues with the aspect ratios of videos on pages as well as errors in the figure/figure caption.

AdarshBhandari15 committed:
Remove duplicate videos

AdarshBhandari15 commented:
Resolves an issue relating to duplicate videos with the same id appearing multiple times. Instead of checking for just the first video (which caused videos to be duplicated with the same id), we now check for the first video that is present in the original npr page's layout. Testing is the same as listed above.

areynold committed:
format publish mapper
